### ISSUE TITLE
Expand generative panel

### ISSUE THREAD

**Issue by emilienschultz** [emilienschultz/activetigger#163]
In collaboration with @JbIPS, the generative panel is going to be refactored

## General philosophy 

Discussed with @eollion 

A panel allowing the user to use third-party end points to annotate text elements with a prompt :

- Configure the third-party end point
- Expertimental step : Enter a prompt and experiment with its results on a few elements (type of inference, format of the result) + Rules to check the formating of the inference (check list : JSON format, regex extraction, etc.)
- Complete step Launch a complete inference on a specific sample (all elements, a number of elements, all untagged elements, etc.) + manage the process (possibility to stop it, prevision of time, error management)

## Identified Tasks

- [x] Add openai endpoint + key
- [ ] Prototype of a design for the frontend separating experimentation / complete prediction

(to be updated)

**Comment by JbIPS** [comment]
I just found a project that could be an inspiration for models integration: https://github.com/open-webui/open-webui

I'll look into it and use any piece of code that could help for OpenAI integration.

**Comment by emilienschultz** [comment]
Mmmh not a bad idea to have a chat-like section for the exploration part, using such connector, for people to try their prompt etc. with their API, before launching the process on the complete dataset.

**Comment by emilienschultz** [comment]
First test of the panel : remarks & comments

Works for ollama (local endpoint) and openai

- [ ] bug due to the change dict->attribute to fix
- [ ] need of a spinner during the process + refresh the page once it is finished$
- [ ] for openai, only put the result of the inference in the Answer field ,not all the answer
- [ ] introspection of existing models (possible to see what were the user choice, either edit or just see it)
- [ ] take a time to explain how to add a new model (e.g. Mistral)
- [ ] to brainstorm : how to save prompts see #259

**Comment by emilienschultz** [comment]
Follow up with the closing discussion  : 
- the structure is set up to add more models (generate methods)
- brainstorming : not a priority to have a prompt repertory
- for the moment, the design is still experimental (no separation between experimental / systematic)
